rasa/core/policies/rule_policy.py

```diff
@@ -75,6 +93,7 @@ def __init__(
         core_fallback_threshold: float = 0.3,
         core_fallback_action_name: Text = ACTION_DEFAULT_FALLBACK_NAME,
         enable_fallback_prediction: bool = True,
+        restrict_rules: bool = True,
```

Comment:
As the parameter name is user interfacing, I think this name would make it clearer what the parameter is doing.
```suggestion
        restrict_rule_length: bool = True,
```

Reply:
I'm not sure, since it doesn't restrict the length of the rule. It can contain arbitrary number of actions

Reply:
Also, we might update the restriction if we get feedback that current restriction is bad

Reply:
> Also, we might update the restriction if we get feedback that current restriction is bad

I think we shouldn't write code for things which **maybe** will happen. Often enough this doesn't happen and then you're stuck with the complexity for no reason.

> I'm not sure, since it doesn't restrict the length of the rule. It can contain arbitrary number of actions

Good point 🤔 
@b-quachtran Do you have suggestions / objections regarding keep `restrict_rules`?

Reply:
> I think we shouldn't write code for things which maybe will happen.
ok

Reply:
anyhow, up for renaming

Reply:
Does it make sense to turn it around? `restrict` sounds like we are limiting something. Users might think they "unleash" the RulePolicy by setting this to `False`. Should we rather call the default something like  `safe_mode` ?

Reply:
Do you mean to create a constant with a string `"safe_mode"`?

Reply:
or creating `safe_mode: True` option?

Reply:
> or creating safe_mode: True option?

This one

Reply:
i honestly have no strong opinions on this. but i don't see any issues with `restrict_rules` tbh. `restrict_rule_length` kind of does make sense, even if it only restricts the number of user messages and not the number of actions. 
I would say up to you

Reply:
Ok, then let's leave it as it is 👍